Wikipedia/Code/InsertMediaSettingsViewController.swift

```diff
@@ -270,14 +273,26 @@ final class InsertMediaSettingsViewController: ViewController {
         do {
             let wikitextWithImage = try WKWikitextUtils.insertImageWikitextIntoArticleWikitextAfterTemplates(imageWikitext: imageWikitext, into: wikitext)
             goToEditPreview(with: wikitextWithImage)
-        } catch {
-            print("Error preparing wikitext")
+        } catch let error {
+            print("Error preparing wikitext\(error)")
         }
 
     }
 
     func goToEditPreview(with wikitext: String) {
-        print(wikitext)
+        guard let articleURL else {
+            print("Error")
+            return
+        }
+
+        let editPreviewViewController = EditPreviewViewController(pageURL: articleURL)
+        editPreviewViewController.theme = theme
+        editPreviewViewController.sectionID = sectionNumber
+        editPreviewViewController.languageCode = articleURL.wmf_languageCode
+        editPreviewViewController.wikitext = wikitext
+        editPreviewViewController.delegate = self
+
+        navigationController?.pushViewController(editPreviewViewController, animated: true)
```

Comment:
I notice the preview content is hidden beneath the navigation bar. I'm happy to merge now and we fix as a followup, just wanted to call it out. It's probably our custom navigation bar up to its old tricks again.

![screenshot23](https://github.com/wikimedia/wikipedia-ios/assets/3620196/854b928f-96b5-4052-a666-f393098f9da4)

Reply:
If you don't mind merging as-is, I'll take a look on it after wrapping up the next task